marketplace_builder/views/pages/tutorials/performance/cache.liquid

```diff
@@ -0,0 +1,106 @@
+---
+converter: markdown
+metadata:
+  title: Cache 
+  description: Detailed descriptions of page, action, and fragment caching with examples.  
+slug: tutorials/performance/cache
+searchable: true
+---
+
+## Page caching
+
+Page caching means you create a static HTML site, and serve it bypassing the whole server side, including authorization. It's super fast, but you can rarely use it. For example, it's great to use this for the home page, to make it very fast for non-authenticated users, but you can't do the same for a dashboard, because the user has to be authenticated. 
+
+**Example:** 
+
+```liquid
+{% raw %}
+cache_static_page_for: 0 
+{% endraw %}
+```
+{% include 'alert/note', content: 'In the old infrastructure of PlatformOS, you used `cache_for`, in the new infrastructure you should use `cache_static_page_for`.' %}
+ 
+## Action caching
+
+Action caching works similarly to page caching, you can use it to cache the whole page, but the difference is that it triggers authorization, so it works also for logged in users.  
+
+**Example:** 
+
+```liquid
+{% raw %}
+cache_expiration_key > 
+	{{ context.page.slug }} 
+cache_action_for: 10 
+cache_layout: true 
+{% endraw %}
+```
+
+**Explanation:** 
+
+There are three properties that control the action cache: 
+
+* The **`cache_expiration_key`** allows you to define a unique identifier for the page. By default, it's the full page URL. If you cache a page for 10 seconds, and the `cache_expiration_key` is blank, cache will be generated for the whole URL. If the user changes anything in the URL, the cache will not be taken into consideration. 
+
+   In a scenario, where you have multiple product pages, and you want to cache the product page, you can have a GraphQL query here: 
+   
+   ```liquid
+   {% raw %}
+   cache_expiration_key > 
+	   {% graph_query 'get_cached_at_from_product', result_name: 'g' %} 
+	   {{ context.page.slug }}/{{ g.results.first.cached_at }}
+   cache_action_for: 0 
+   cache_layout: true 
+   {% endraw %}
+   ```
+
+* The default for `cache_action_for` is as long as possible. Setting it to `0` means invalidate the cache if anything changes (i.e. when the `cache_expiration_key` changes).  
+* The default for `cache_layout` is `false`. 
+
+## Fragment caching
+
+Using page caching, or action caching for the whole page, whenever you change something on the page and deploy it, the cache will be invalidated. In reality, it's rare, that you would want to cache the whole page or layout (this is why `cache_layout` is `false` by default). 
```

Comment:
> In reality, it's rare, that you would want to cache the whole page

actually that's what everyone wants, but rarely it is possible because of personalization - different content for different users